### ISSUE TITLE
AJAX Submit Not Working on IE9 & below

### ISSUE THREAD

**Issue by designcise** [jquery-form/form#438]
The problem: When submitting form on IE<=9, the result returns an entire page's HTML code -- this is with no upload file selected. It, more or less, has the same effect with a file selected. Removing enctype="multipart/form-data" attribute on the form element makes the form work fine, but then obviously the upload doesn't work.

I've spent hours looking for a solution to this; a working example would be much appreciated! I understand that IE can't do AJAX and must therefore use the iFrame method, but can someone please point me to a working example? Ones I've found are either too ambiguous or are hard to follow. Thanks!

**Comment by NullDivision** [comment]
I'm having the same issue. IE supports AJAX submit just fine. The problem however is submitting files which, since IE lacks the FileAPI reverts to a piece of legacy code.

Specifically in cb() there's a piece of code:

```
if (docRoot) {
    xhr.status = Number( docRoot.getAttribute('status') ) || xhr.status;
    xhr.statusText = docRoot.getAttribute('statusText') || xhr.statusText;
}
```

As you can see it tries to get the information from the document root but apparently this never gets a response correctly and thus returns the entire iFrame body.

**Comment by designcise** [comment]
I managed to resolve it by:
1. Changing the content type header to header('content-type: text/plain');
2. Responding to the call for IE in non-AJAX mode (by checking for $_SERVER['HTTP_X_REQUESTED_WITH'])
3. Setting the "dataType: 'json'"

**Comment by NullDivision** [comment]
What do you mean by "non-AJAX mode"?

**Comment by designcise** [comment]
I meant, since the hack used for IE 9 is to use iframe (which is essentially not an AJAX call to the server), you need to respond to the request accordingly. The way to check for an ajax call is to usually check the "$_SERVER['HTTP_X_REQUESTED_WITH']" value. If it is an AJAX call you ignore it in case of IE, and if it's a non-AJAX call you return a response (which is loaded in the iframe).

There's some information regarding that here: http://malsup.com/jquery/form/#file-upload

**Comment by bq-hentai** [comment]
Hmmm, I met a problem the same as this (a very very old project). I checked nearly all issues and find some in stackoverflow, but I can't find an 'official' answer.

The version of the plugin is `version: 3.49.0-2014.02.05`. And when upload file in browser <= IE9, the response is html string.

Can anyone actually solve this? Or the latest version solved this?

**Comment by designcise** [comment]
I've been using the workaround I posted; if memory serves right it works by: 

1. Checking if the browser is IE (for example, by sending a flag from client side to server side script; which is set `true` for ie, and `false` otherwise);
2. Set the content header types based on whether ajax call came from an IE <= 9 browser or not, for example: `header('content-type: ' . (($is_ie) ? 'text/plain' : 'application/json') . '; charset=utf-8');`

Hope that helps!

**Comment by bq-hentai** [comment]
@designcise Thx for ur reply.

Now the content-type in response header is 'text/html'。This method is used to prevent IE download maybe?

**Comment by bq-hentai** [comment]
@designcise my teammate test this situation —— upload with a form without using jq-form. And then... fail.
